ts/Dashboards/Component/KPIComponent.ts

```diff
@@ -349,11 +349,73 @@ class KPIComponent extends Component {
         }
         return '';
     }
+    public toJSON(): KPIComponent.ClassJSON {
+        const base = super.toJSON();
+        const json = {
+            ...base,
+            options: {
+                ...base.options,
+                value: this.options.value,
+                subtitle: JSON.stringify(this.options.subtitle),
+                title: JSON.stringify(this.options.title),
+                threshold: this.options.threshold,
+                style: JSON.stringify(this.options.style),
+                thresholdColors: this.options.thresholdColors,
+                chartOptions: JSON.stringify(this.options.chartOptions),
+                valueFormat: this.options.valueFormat
+            }
+        };
+
+        this.emit({ type: 'toJSON', json: base });
+
+        return json;
+    }
+
+    public static fromJSON(json: KPIComponent.ClassJSON): KPIComponent {
+        const options = json.options;
+        const chartOptions =
+            options.chartOptions && JSON.parse(options.chartOptions);
+        const subtitle = JSON.parse(options.subtitle || '{}');
+        const title = options.title && JSON.parse(options.title);
+        const style = JSON.parse(options.style || '{}');
+        const thresholdColors = options.thresholdColors;
+        const value = options.value;
+        const threshold = options.threshold;
+        const valueFormat = options.valueFormat;
+
+        return new KPIComponent(
+            merge(options, {
+                chartOptions,
+                title,
+                subtitle,
+                style,
+                thresholdColors,
+                value,
+                threshold,
+                valueFormat
+            })
+        );
+    }
 }
 
 namespace KPIComponent {
     export type ComponentType = KPIComponent;
 
+    export interface ClassJSON extends Component.JSON {
+        options: ComponentJSONOptions;
+
+    }
+
+    export interface ComponentJSONOptions extends Component.ComponentOptionsJSON {
+        title?: string;
+        chartOptions?: string;
```

Comment:
Why does chartOptions get stringified and not assigned directly with `Core/Options` type? `import type` is safe as it does not add to the Dashboard bundle.

Reply:
I looked at how it is done in the `HighchartsComponent` and wanted to be consistent. I can change it though, if you like

Reply:
I see. This is probably related to the callback options to Highcharts. Let's keep it for now.